tooling/scripts/prepare_release.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+
+from dataclasses import dataclass
+from enum import Enum
+import os
+import re
+import xml.etree.ElementTree as et
+from datetime import datetime
+import git
+
+
+# Filter and convert tags to version numbers
+def extract_version(tag_name):
+    # Strip 'v' prefix if present
+    version_str = tag_name.lstrip('v')
+    # Match the major, minor and patch number
+    match = re.match(r'^(\d+)\.(\d+)\.(\d+)$', version_str)
+    if match is None:
+        raise ValueError(f'Invalid version string, '
+                         f'must be int.int.int: "{version_str}"')
+    # Cast all numbers to integers
+    version = [int(x) for x in match.groups()]
+    return version
+
+
+# Find the tag with the highest version number in a list of git tags
+def extract_highest_version(tag_list):
+    # Filter out tags that don't have valid versions
+    versioned_tags = []
+    for tag in tag_list:
+        try:
+            version = extract_version(tag.name)
+            versioned_tags.append((tag, version))
+        except ValueError:
+            pass
+    # Sort tags by their version numbers
+    versioned_tags.sort(key=lambda x: x[1], reverse=True)
+    # Return the highest tag or None
+    return versioned_tags[0][0] if versioned_tags else None
+
+
+# Increment a version number by a given bump level
+def bump_version(version, level='patch'):
+    # Extract version number
+    bumped_version = extract_version(version)
+    # Increment appropriate part
+    bump_idx = {'major': 0, 'minor': 1, 'patch': 2}[level]
+    bumped_version[bump_idx] += 1
+    # Reset trailing parts
+    bumped_version = [
+        x if idx <= bump_idx else 0 for idx, x in enumerate(bumped_version)
+    ]
+    return bumped_version
+
+
+# Types of packages we support
+class PkgType(Enum):
+    CMake = 1
+    ROS1 = 2
+
+
+# Class used to specify a package in our repository
+@dataclass
+class Pkg:
+    name: str
+    type: PkgType
+
+
+# Determine the relative path to a package given its name and type
+def determine_package_path(repository_path, package):
+    if package.type is PkgType.CMake:
+        sub_paths = ['library/cpp', 'library', 'libraries']
+        return [
+            os.path.join(repository_path, sub_path) for sub_path in sub_paths
+        ]
+    if package.type is PkgType.ROS1:
+        sub_paths = ['interfaces/ros1', 'ros']
+        return [
+            os.path.join(repository_path, sub_path, pkg.name)
+            for sub_path in sub_paths
+        ]
+
+    raise SystemExit
+
+
+# Parameters
+bump_level = 'major'
+repo_path = '/home/victor/catkin_ws/src/wavemap'
```

Comment:
Specific to your system, not sure if it's worth making this general.

Reply:
Thanks! Will update the code to detect this automatically.